tests/core/providers/test_websocket_provider.py

```diff
@@ -8,6 +21,40 @@
 )
 
 
+@pytest.yield_fixture
+def start_websocket_server(open_port):
+    event_loop = asyncio.new_event_loop()
+
+    def run_server():
+        async def empty_server(websocket, path):
+            data = await websocket.recv()
+            await asyncio.sleep(0.1)
```

Comment:
*if* the structure of the test is such that this sleep duration won't actually effect test run times (in success conditions) then this doesn't need to be changed, however, we've found `0.1` is quite a long time during test runs.  Consider dropping this to something like `0.02`.